tools/cldr-apps/src/main/java/org/unicode/cldr/web/MailSender.java

```diff
@@ -259,13 +262,16 @@ public void queue(Integer fromUser, int toUser, String subject, String body, CLD
             xpath = null;
         }
         final DBUtils dbutils = DBUtils.getInstance();
+        Connection conn = dbutils.getDBConnection();
+        if (conn == null) {
+            return;
+        }
         try (
```

Comment:
seems to be cause of CLDR-16179 - Connection does not get cleaned up and so the app starves for connections

Reply:
so the clean-up is performed due to getDBConnection being called within parentheses of try(), that's clear now

elsewhere getDBConnection is often used differently, typically with explicit close() as in

        } finally {
            DBUtils.close(s, s2, s3, conn);
        }

I'm going to check the rest of this old pr for other places that may need fixing

Would it be OK with you if getDBConnection never returned null, but threw an exception instead? I can't help paying attention to compiler warnings, and checking for null seems very inconsistent with the various ways the method is currently actually used